cooked-validators/src/Cooked/MockChain/Monad.hs

```diff
@@ -134,8 +134,8 @@ spOutResolveDatum ::
   MonadBlockChain m =>
   SpendableOut ->
   m SpendableOut
-spOutResolveDatum (txOutRef, chainIndexTxOut@(Pl.ScriptChainIndexTxOut _ _ (Left _) _)) = do
-  mDatum <- datumFromTxOut chainIndexTxOut
+spOutResolveDatum (txOutRef, chainIndexTxOut@(Pl.ScriptChainIndexTxOut _ _ (Left datumHash) _)) = do
```

Comment:
Not related to the changes in this PR, but function `spOutResolveDatum`. For now, it assumes resolving the datum may only be required in `ScriptChainIndexTxOut`. Are we sure it is never necessary for public key outputs? For now in that case the function returns the spOut unchanged.